### PR TITLE
Don't assume ScalaTest run cunfiguration

### PR DESCRIPTION
**Pull request by tpasternak**
When configuring BspFetchTestEnvironmentTask, it should not be assumed
the runconfig is ScalaTest

### CODE REVIEW

File: bsp/src/org/jetbrains/bsp/project/test/environment/BspFetchTestEnvironmentTaskProvider.scala [hunk 1]
```diff
@@ -60,12 +60,16 @@ class BspFetchTestEnvironmentTaskProvider extends BeforeRunTaskProvider[BspFetch
...
 
   override def configureTask(context: DataContext, configuration: RunConfiguration, task: BspFetchTestEnvironmentTask): Promise[lang.Boolean] = {
-    val module = configuration.asInstanceOf[ScalaTestRunConfiguration].getModule
-    val res = for {
-      potentialTargets <- getBspTargets(module)
-      _ <- askUserForTargetId(potentialTargets.map(_.getUri), task)
-    } yield ()
-    Promises.resolvedPromise(res.isDefined)
+    configuration match {
+      case moduleBasedConfiguration: ModuleBasedConfiguration[_,_] =>
```
Comment (unkarjedy) [inline bsp/src/org/jetbrains/bsp/project/test/environment/BspFetchTestEnvironmentTaskProvider.scala:64]: If it's only about tests,  maybe we should only handle `AbstractTestRunConfiguration`, which is base for all supported test frameworks? Not to "pollute" all unrelated configurations.

**Review by jastice — APPROVED** [review]
thanks!

**Comment by jastice** [comment]
merged/rebased via build server: https://github.com/JetBrains/intellij-scala/commit/e59062ecfbfb90804080b33d0a3c8cde2f65222b

**Comment by tpasternak** [comment]
thanks!

**Comment by unkarjedy** [comment]
turned out I again didn't press "Submit review"... 🤦‍♂️
who designed this GitHub?)

File: bsp/src/org/jetbrains/bsp/project/test/environment/BspFetchTestEnvironmentTaskProvider.scala (hunk 1 above)
Comment (tpasternak) [reply bsp/src/org/jetbrains/bsp/project/test/environment/BspFetchTestEnvironmentTaskProvider.scala:64]: But then you couldn't support JUnit

Comment (unkarjedy) [reply bsp/src/org/jetbrains/bsp/project/test/environment/BspFetchTestEnvironmentTaskProvider.scala:64]: ah, I see, so maybe add `if BspTesting.isBspRunnerSupportedConfiguration`

Comment (tpasternak) [reply bsp/src/org/jetbrains/bsp/project/test/environment/BspFetchTestEnvironmentTaskProvider.scala:64]: Yeah, that might be good, but we already check this before adding task to the run config https://github.com/JetBrains/intellij-scala/blob/33de905217cc97759d367c2659a2ee1352baa8ed/bsp/src/org/jetbrains/bsp/project/test/environment/BspFetchTestEnvironmentTaskInstaller.scala#L22

**Pull request closed without merging**